src/biogeophys/TridiagonalMod.F90

```diff
@@ -70,27 +68,13 @@ subroutine Tridiagonal (bounds, lbj, ubj, jtop, numf, filter, a, b, c, r, u)
     do j = lbj, ubj
        do fc = 1,numf
           ci = filter(fc)
-          if ((col%itype(ci) == icol_sunwall .or. col%itype(ci) == icol_shadewall &
-              .or. col%itype(ci) == icol_roof) .and. j <= nlevurb) then
```

Comment:
- [ ] It looks like the original code avoided setting values at some indices that were outside the level bounds of the current column, whereas the new code sets values for all indices (presumably down to nlevmaxurbgrnd) for all columns. @olyson , was this change intentional? I see a similar change in a few other places, which I will call out in separate comments.

Reply:
Yes, it was intentional.  Tridiagonal and TridiagonalCol are called from within a hydrologyc filter loop.  Sunlit wall, shaded wall, and roof are hydrologically inactive (except for roof which can have surface water/snow) and are excluded from the hydrologyc filter.  So I decided that it was unnecessary to have this if statement in the subroutines.

Reply:
That makes sense - thanks!